### PR TITLE
#315: Set up the Dockerfile for Backend

### PR DESCRIPTION
**Pull request by pawelcieszko**

### CODE REVIEW

File: src/main/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM openjdk:10
```
Comment (pgesek) [inline src/main/docker/Dockerfile]: add maintainer and expose entries

File: pom.xml [hunk 2]
```diff
@@ -164,6 +167,30 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>com.spotify</groupId>
+                <artifactId>docker-maven-plugin</artifactId>
+                <version>${dockerfile-maven-plugin.version}</version>
+                <configuration>
+                    <imageName>${docker.image.prefix}/${project.artifactId}</imageName>
+                    <dockerDirectory>src/main/docker</dockerDirectory>
```
Comment (pgesek) [inline pom.xml]: Personally I prefer to store the dockerfile in the root of the app, which is the more widespread convention. Any reason to keep in src/main/docker?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        depends_on:
+            - postgresql
+    postgresql:
+        image: postgres:9.5.12
+        environment:
+        - POSTGRES_PASSWORD=mysecretpassword
+        - POSTGRES_USER=postgres
+        - POSTGRES_DB=mathshare
+        volumes:
+        - db-data:/var/lib/postgresql/data
```
Comment (dserkowski) [inline docker-compose.yml]: There is something wrong with indentations

File: pom.xml (hunk 2 above)
Comment (pawelcieszko) [reply pom.xml]: There is no specific reason. I have found this convention in several Spring Boot + Docker application. However, I will try to use the root of the app.

File: docker-compose.yml (hunk 3 above)
Comment (pawelcieszko) [reply docker-compose.yml]: Right, I will fix it

File: src/main/docker/Dockerfile (hunk 1 above)
Comment (pawelcieszko) [reply src/main/docker/Dockerfile]: 1. The MAINTAINER entry is deprecated. Should I use LABEL instead?
https://docs.docker.com/engine/reference/builder/#maintainer-deprecated
2. Who should be added as maintainer (with email)?
3. I will add EXPOSE entry

Comment (pawelcieszko) [reply src/main/docker/Dockerfile]: Ok I will send an email and add label addition to ticket scope

**Pull request merged**